packages/jest-runner/src/run_test.js

```diff
@@ -151,6 +184,8 @@ async function runTestInternal(
     });
   } finally {
     await environment.teardown();
+
+    sourcemapSupport.resetRetrieveHandlers();
```

Comment:
seems like this is too early - I think the uninstall needs to happen after test results are collected